examples/tools/generate.ipynb

```diff
@@ -0,0 +1,514 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Intensity-based samples"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Data samples are more complicated than phase space samples in that they\n",
+    "represent the intensity profile resulting from a reaction. You therefore need\n",
+    "an :class:`.Intensity` object and a phase-space over which to generate that\n",
+    "intensity distribution. We therefore call such a data sample an\n",
+    "**intensity-based sample**.\n",
+    "\n",
+    "An intensity-based sample is generated with the function\n",
+    ":func:`~.ui.generate`. Its usage is similar to :func:`.generate_phsp`, but\n",
+    "now you have to give an :class:`.Intensity` and a :class:`.Kinematics`\n",
+    "instance as additional arguments. Sampling is performed through hit & miss,\n",
+    "which is reflected by the efficiency printed at the end."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create intensity profile\n",
+    "intensity_builder = pwa.IntensityBuilderXML(\n",
+    "    'model.xml', particle_list, kinematics, phsp_sample)\n",
+    "intensity = intensity_builder.create_intensity()\n",
+    "\n",
+    "# generate data\n",
+    "data_sample = pwa.generate(\n",
+    "    100, kinematics, generator, intensity, random_generator)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importance sampling"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "In the example above, we used the decay $J/\\psi \\to \\gamma \\pi^0 \\pi^0$\n",
+    "with an f resonances only. Imagine, however, that we want to investigate the\n",
+    "resonance $\\omega \\to \\gamma\\pi^0$. This resonance has such a narrow peak\n",
+    "that you will run into problems when normalizing the intensity model.\n",
+    "\n",
+    "When normalizing, you need to numerically compute an integral over the\n",
+    "intensity distribution with a *hit and miss strategy*, but with a narrow\n",
+    "peak, you will have enough hits within the peak (where the contribution to\n",
+    "the integral is highest). You therefore need to have a high density of\n",
+    "phase-space points near the resonance.\n",
+    "\n",
+    "Now, you could just use an evenly distributed phase space sample like the one\n",
+    "we generated above with a much larger number of data points, but this will\n",
+    "result in many hits outside the peak (the region that hardly contributes to\n",
+    "the integral). A smarter strategy is generate a sample that has a higher\n",
+    "density of phase space points within the peak region. This strategy is called\n",
+    "**importance sampling**."
```

Comment:
Bit better than before but still confusing and not precise. MC based numerical integration does not do hit and miss but just sums up the function values using the phase space sample. Now in order to have a good estimate for a narrow peak structure, you need either a very large unweighted phase space sample, or a weighted phase space sample with more events in the peak region (but overall smaller sample size).

Reply:
https://github.com/ComPWA/pycompwa/pull/79/commits/1344f04f13a30904bc66b2680c7857824ec116ad